### PR TITLE
fix(api-client): prefill auth

### LINKED ISSUE #3377: On 1.25.27, prefilled token no longer get passed to the "Test Request" dialog
### What happens?

Scalar allows setting a token in the `configuration` object to pre-fill the credentials for all API requests. For e.g

```javascript
var configuration =  {
   authentication: {
      http: {
            bearer: {token: getToken()},
            basic: {}
      },
   }
}
```

This seems to be broken on v1.25.27, where opening the "Test Request" dialog no longer shows the pre-filled token, or use the pre-filled token when making request. This still works for v1.25.0 at least.

This issue also seems to be related to this one: https://github.com/scalar/scalar/issues/3168.

### What did you expect to happen?

"Test Request" dialog should use pre-filled token from the configuration object.

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by amritk**
Closes #3377

1. Pre-fills client auth according to the authentication config object.
2. Syncs up the collection selected auth with the references one.
3. Update the selectedSecuritySchemeUids to come from the collection on the modal, and the request otherwise

This should hold us over until we get a collection page

### CODE REVIEW

File: packages/api-client/src/layouts/Modal/create-api-client-modal.ts [hunk 1]
```diff
@@ -36,12 +36,14 @@ export const createApiClientModal = async (
     await importSpecFromUrl(configuration.spec.url, 'default', {
       proxy: configuration.proxyUrl,
       overloadServers: configuration?.servers,
-      preferredSecurityScheme: configuration.preferredSecurityScheme,
+      authentication: configuration.authentication,
```
Comment (amritk) [inline packages/api-client/src/layouts/Modal/create-api-client-modal.ts:39]: we pass through the whole auth state now

File: packages/api-client/src/store/import-spec.ts [hunk 2]
```diff
@@ -16,6 +15,15 @@ export const specDictionary: Record<
   { hash: number; schema: OpenAPIV3.Document | OpenAPIV3_1.Document }
 > = {}
 
+type ImportSpecFileArgs = ImportSpecToWorkspaceArgs & {
```
Comment (amritk) [inline packages/api-client/src/store/import-spec.ts:18]: Just re-using types

File: packages/api-client/src/views/Request/RequestSection/RequestAuth/RequestAuth.vue [hunk 3]
```diff
@@ -58,22 +58,23 @@ const teleportId = `combobox-${nanoid()}`
...
-    isReadOnly.value && requestSecurity?.length
-      ? requestSecurity.map((s) => {
-          const nameKey = Object.keys(s)[0]
-          return (
-            Object.values(securitySchemes).find((ss) => ss.nameKey === nameKey)
-              ?.uid ?? ''
-          )
-        })
-      : activeCollection.value?.securitySchemes
+  // const requestSecurity = activeRequest.value?.security?.filter(
```
Comment (amritk) [inline packages/api-client/src/views/Request/RequestSection/RequestAuth/RequestAuth.vue:63]: keep this in for now, we will use it for https://github.com/scalar/scalar/issues/3613

File: packages/api-reference/index.html [hunk 4]
```diff
@@ -19,10 +19,29 @@
     <!-- Optional: You can set a full configuration object like this: -->
     <script>
       var configuration = {
-        // defaultHttpClient: {
-        //   targetKey: 'node',
-        //   clientKey: 'undici',
-        // },
+        authentication: {
```
Comment (amritk) [inline packages/api-reference/index.html:22]: This is for testing, can remove before merging

**Comment by amritk** [comment]
@coderabbitai review

**Review by marclave — APPROVED** [review]
unreal, great fix @amritk

**Pull request merged**